### DISCUSSION TITLE
Importing certificates from Machine ID into web browsers

### DISCUSSION THREAD

**Discussion by Aloysius-AS** [gravitational/teleport#29664] [Q&A]
Hi all,

I am trying to leverage on Machine ID to enable web browser to access web application protected by Teleport. Machine ID is configured on a Ubuntu host.

First off, I noted that the certificates cannot be imported natively into the browser; they need to be converted into a single PKCS#12 file before import is possible in Firefox.
openssl pkcs12 -export -in tlscert -inkey key -out tlscert.p12

As Machine ID generates certificates on a periodic frequency, is there a way for the certificates to be automatically imported into web browsers?

Thank you all for any help you can give.

**Comment by webvictim** [comment]
This isn't a use case that we've considered, because going to the application's Teleport public address in the browser will make you authenticate with Teleport in the browser, then set cookies to authenticate you directly to the application without the need for a client certificate.

Presumably you're trying to automate the login process to enable some kind of browser integration?

cc @strideynet

**Comment by Aloysius-AS** [comment]
Thank you for your insight.

Yes, you are right; I am exploring browser integration to automate the login process to Teleport applications.
